src/aac/template_engine.py

```diff
@@ -120,18 +120,34 @@ def write_generated_templates_to_file(
         generated_files: list of generated files to write to the filesystem
         output_directory: the directory to write the generated files to.
     """
-    if not os.path.exists(output_directory):
-        os.mkdir(output_directory)
 
+    _ensure_directory_exists(output_directory)
     for generated_file in generated_files:
         _write_file(
-            output_directory,
+            _full_output_directory(output_directory, generated_file),
             generated_file.file_name,
             generated_file.content,
             generated_file.overwrite,
         )
 
 
+def _full_output_directory(output_directory: str, generated_file: TemplateOutputFile) -> str:
```

Comment:
Function names should incorporate verbs for better clarity.
```suggestion
def _get_template_output_directory(output_directory: str, generated_file: TemplateOutputFile) -> str:
```

Reply:
Resolved with ab43e25